examples/custom-logging/index.ts

```diff
@@ -0,0 +1,21 @@
+import { TypeScriptGenerator, OutputModel, ModelLoggingInterface, Logger } from '../../src';
+
+const doc = {
+  $id: 'Address',
+  type: 'object',
+  properties: {
+    street_name: { type: 'string' },
+    city: { type: 'string', description: 'City description' },
+    house_number: { type: 'number' },
+    marriage: { type: 'boolean', description: 'Status if marriage live in given house' },
+    pet_names: { type: 'array', items: { type: 'string' } },
+    state: { type: 'string', enum: ['Texas', 'Alabama', 'California', 'other'] },
+  },
+  required: ['street_name', 'city', 'state', 'house_number', 'state'],
+};
+
+export async function generate(logger: ModelLoggingInterface) : Promise<OutputModel[]> {
+  Logger.setLogger(logger);
```

Comment:
for logging example I would expect to actually see an example of ModelLoggingInterface

Reply:
I thought about doing that, but I could not figure out to write the test to verify it works if I can't control the input 🤔?

Reply:
what do you mean you can't control the input? what input, you have a hardcoded doc here, or am I missing something? 🤔

Reply:
I use the jest mocked functions to ensure that the logger actually logs something - https://github.com/asyncapi/modelina/pull/338/files/7e6e530c96a65d52c7d6159b18181bf71c190f78#diff-582bf3a7b78962a1559eab563f529c43fe057b9519bde459e24d2f8d47678c8c

Reply:
But of course, this comes down to, what exactly am I asserting or ensuring works inside the example tests and I am not sure how to balance it 😬